compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/interop/UIKitView.uikit.kt

```diff
@@ -260,91 +305,54 @@ fun <T : UIViewController> UIKitViewController(
-    val interopContext = LocalUIKitInteropContext.current
-
-    EmptyLayout(
-        modifier.onGloballyPositioned { coordinates ->
-            localToWindowOffset = coordinates.positionInRoot().round()
-            val newRectInPixels = IntRect(localToWindowOffset, coordinates.size)
-            if (rectInPixels != newRectInPixels) {
-                val rect = newRectInPixels.toRect().toDpRect(density)
-
-                interopContext.deferAction {
-                    embeddedInteropComponent.wrappingView.setFrame(rect.asCGRect())
-                }
-
-                if (rectInPixels.width != newRectInPixels.width || rectInPixels.height != newRectInPixels.height) {
-                    interopContext.deferAction {
-                        onResize(
-                            embeddedInteropComponent.component,
-                            CGRectMake(0.0, 0.0, rect.width.value.toDouble(), rect.height.value.toDouble()),
-                        )
-                    }
-                }
-                rectInPixels = newRectInPixels
-            }
-        }.drawBehind {
-            // Clear interop area to make visible the component under our canvas.
-            drawRect(Color.Transparent, blendMode = BlendMode.Clear)
-        }.trackUIKitInterop(interopContainer, embeddedInteropComponent.wrappingView).let {
-            if (interactive) {
-                it.then(InteropViewCatchPointerModifier())
-            } else {
-                it
-            }
-        }.interopSemantics(accessibilityEnabled, embeddedInteropComponent.wrappingView)
+    UIKitInteropLayout(
+        modifier = modifier,
+        update = update,
+        background = background,
+        entityHandler = handler,
+        onResize = onResize,
+        interactive = interactive,
+        accessibilityEnabled = accessibilityEnabled
     )
-
-    DisposableEffect(Unit) {
-        embeddedInteropComponent.component = factory()
-        embeddedInteropComponent.updater = Updater(embeddedInteropComponent.component, update) {
-            interopContext.deferAction(action = it)
-        }
-
-        interopContext.deferAction(UIKitInteropViewHierarchyChange.VIEW_ADDED) {
-            embeddedInteropComponent.addToHierarchy()
-        }
-
-        onDispose {
-            interopContext.deferAction(UIKitInteropViewHierarchyChange.VIEW_REMOVED) {
-                embeddedInteropComponent.removeFromHierarchy()
-            }
-        }
-    }
-
-    LaunchedEffect(background) {
-        interopContext.deferAction {
-            embeddedInteropComponent.setBackgroundColor(background)
-        }
-    }
-
-    SideEffect {
-        embeddedInteropComponent.updater.update = update
-    }
 }
 
-private abstract class EmbeddedInteropComponent<T : Any>(
+/**
+ * An abstract class responsible for hiearchy updates and state management of interop entities like [UIView] and [UIViewController]
+ */
+private abstract class InteropEntityHandler<T : Any>(
```

Comment:
Introducing another abstract word "entity" doesn't help. `InteropComponent` was better

Reply:
Agree on Entity against Component, but will better keel the `Handler` part. Because it's not _the_ component (UIView or UIViewController) it's something helping us to manage it